app/scripts/lib/multichain-api/caip-permission-adapter-middleware.test.js

```diff
@@ -0,0 +1,175 @@
+      await handler(baseRequest);
+      expect(next).toHaveBeenCalled();
+    });
+
+    it('does not read the permission state', async () => {
+      const { handler, getCaveat } = createMockedHandler();
+      await handler(baseRequest);
+      expect(getCaveat).not.toHaveBeenCalled();
+    });
+  });
+
+  describe('BARAD_DUR feature flag is set', () => {
+    beforeAll(() => {
+      process.env.BARAD_DUR = 1;
+    });
+
+    it('gets the authorized scopes from the CAIP-25 endowement permission', async () => {
+      const { handler, getCaveat } = createMockedHandler();
+      await handler(baseRequest);
+      expect(getCaveat).toHaveBeenCalledWith(
+        'http://test.com',
+        Caip25EndowmentPermissionName,
+        Caip25CaveatType,
+      );
+    });
+
+    it('allows the request when there is no CAIP-25 endowement permission', async () => {
+      const { handler, getCaveat, next } = createMockedHandler();
+      getCaveat.mockReturnValue(null);
+      await handler(baseRequest);
+      expect(next).toHaveBeenCalled();
+    });
+
+    it('allows the request when the CAIP-25 endowement permission was not granted from the multichain flow', async () => {
+      const { handler, getCaveat, next } = createMockedHandler();
+      getCaveat.mockReturnValue({
+        value: {
+          isMultichainOrigin: false,
+        },
+      });
+      await handler(baseRequest);
+      expect(next).toHaveBeenCalled();
+    });
+
+    it('gets the chainId for the request networkClientId', async () => {
+      const { handler, getNetworkConfigurationByNetworkClientId } =
+        createMockedHandler();
+      await handler(baseRequest);
+      expect(getNetworkConfigurationByNetworkClientId).toHaveBeenCalledWith(
+        'mainnet',
+      );
+    });
+
+    it('throws an error if the requested scope method is not authorized in either the current scope or the wallet scope', async () => {
+      const { handler, end } = createMockedHandler();
+
+      await handler({
+        ...baseRequest,
+        method: 'unauthorized_method',
+      });
+      expect(end).toHaveBeenCalledWith(providerErrors.unauthorized());
+    });
+
+    it('allows the request if the requested scope method is authorized in the current scope', async () => {
+      const { handler, next } = createMockedHandler();
+
+      await handler(baseRequest);
+      expect(next).toHaveBeenCalled();
+    });
+
+    it('allows the request if the requested scope method is authorized in the wallet scope and the current scope does exist in the authorization', async () => {
+      const { handler, next } = createMockedHandler();
+
+      await handler({
+        ...baseRequest,
+        method: 'wallet_watchAsset',
+      });
+      expect(next).toHaveBeenCalled();
+    });
+
+    it('allows the request if the requested scope method is authorized in the wallet scope and the current scope does not exist in the authorization', async () => {
```

Comment:
oh ok I see. this is an interesting interpretation that we should probably discuss. `wallet_watchAsset` is authorized in the wallet scope and we're saying it can be used against any other scope in the session. Not saying its wrong. Hadn't thougth about this case tho

Reply:
yeah this is a weird one. I remember we talked about this and decided something like either
- `wallet` scoped methods should actually contain a list of CAIP-2 scopes that they're authorized to be called with 
- or they should be authorized under each scope they will be used with

Reply:
Something to discuss with casa. but what are your thoughts on these options?

Reply:
both are a little awkward, but the latter is easier to implement. With this particular method, `wallet_watchAsset`, it makes more sense to associate it with a specific chain/scope, but that isn't necessarily the case with other `wallet_` methods. If we take a look at `wallet_swapAssets`, it's no longer as obvious what we should do. Does specifying `wallet_swapAssets` for a scope mean that we can send from, send to, or both? CAIP-25 authorizations do not let us be this specific

Reply:
Yea agreed. This needs further definition. I agree the latter approach is simpler and probably what we should do for now

Reply:
addressed https://github.com/MetaMask/metamask-extension/pull/26296/commits/0751af2aa64e2b42178dfb9f8dad958dd8bef8b6

Reply:
Asked @bumblefudge about this in CASA discord